flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,285 @@
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+
+        Schema schema =
+                Schema.newBuilder()
+                        .column("f0", DataTypes.STRING())
+                        .column("f1", DataTypes.of(DenseVector.class))
+                        .build();
+
+        DataStream<Row> dataStream = env.fromCollection(trainArray);
+        trainData = tEnv.fromDataStream(dataStream, schema).as("label, vec");
+
+        Schema outputSchema =
+                Schema.newBuilder().column("f0", DataTypes.of(DenseVector.class)).build();
+
+        DataStream<Row> predDataStream = env.fromCollection(testArray);
+        testData = tEnv.fromDataStream(predDataStream, outputSchema).as("vec");
+    }
+
+    /** test knn Estimator. */
+    @Test
+    public void testFitAntTransform() throws Exception {
+        Knn knn =
+                new Knn()
+                        .setLabelCol("label")
+                        .setFeaturesCol("vec")
+                        .setK(4)
+                        .setPredictionCol("pred");
+
+        KnnModel knnModel = knn.fit(trainData);
+        Table result = knnModel.transform(testData)[0];
+
+        DataStream<Row> output = tEnv.toDataStream(result);
+
+        List<Row> rows = IteratorUtils.toList(output.executeAndCollect());
+        for (Row value : rows) {
+            for (Row exp : expectedData) {
+                assert !exp.getField(1).equals(value.getField(0))
+                        || (exp.getField(0).equals(value.getField(1)));
+            }
+        }
+    }
+
+    /** test knn Estimator. */
+    @Test
+    public void testParamsConstructor() throws Exception {
+        Map<Param<?>, Object> params = new HashMap<>();
+        params.put(HasLabelCol.LABEL_COL, "label");
+        params.put(HasFeaturesCol.FEATURES_COL, "vec");
+        params.put(HasK.K, 4);
+        params.put(HasPredictionCol.PREDICTION_COL, "pred");
+        Knn knn = new Knn(params);
+
+        KnnModel knnModel = knn.fit(trainData);
+        Table result = knnModel.transform(testData)[0];
+
+        DataStream<Row> output = tEnv.toDataStream(result);
+
+        List<Row> rows = IteratorUtils.toList(output.executeAndCollect());
+        for (Row value : rows) {
+            for (Row exp : expectedData) {
+                assert !exp.getField(1).equals(value.getField(0))
+                        || (exp.getField(0).equals(value.getField(1)));
+            }
+        }
+    }
+
+    /** test knn as a pipeline stage. */
+    @Test
+    public void testPipeline() throws Exception {
+        Knn knn =
+                new Knn()
+                        .setLabelCol("label")
+                        .setFeaturesCol("vec")
+                        .setK(4)
+                        .setPredictionCol("pred");
+
+        List<Stage<?>> stages = new ArrayList<>();
+        stages.add(knn);
+
+        Pipeline pipe = new Pipeline(stages);
```

Comment:
It is not clear what is the extra benefit of testing Pipeline here. `Pipeline` is tested in `PipelineTest`. Could we remove/refactor this test to be consistent with tests of other algorithms?

Reply:
Ut has been deleted.